INSTALL.md

```diff
@@ -0,0 +1,233 @@
+```
+
+To uninstall Netatalk:
+
+```
+sudo ninja -C build uninstall                           # install Netatalk
+```
+
+To test (requires the `-Dwith-tests=true` flag at configure time):
+
+```
+cd build && meson test
+```
+
+## Configuration flags
+
+When using Meson, to review the options which Meson chose, run:
+
+```
+meson configure
+```
+
+With additional arguments the `meson setup build` command can be used to 
+configure Netatalk according to user preference. All generic options passed to
+this command take the form `-Doption=value`.
+
+For example, to install Netatalk in a specific location:
+
+```sh
+meson setup build -Dprefix=/tmp/install
+```
+
+By default meson enables all Netatalk features if their dependencies are available.
+However, many features can be enabled or disabled at configure time.
+All such options take the form `-Dwith-option=true` or `-Dwith-option=false`:
+
+```
+Option                 Description                                 Default value
+
+with-acls              Enable ACL support                          true
+with-afpstats          Enable AFP statistics via D-Bus             true
+with-cracklib          Enable cracklib support                     true
+with-debug             Enable verbose debug code                   false
+with-dtrace            Enable DTrace probes                        true
+with-embedded-ssl      Enable built-in SSL support                 true
+                       (required for DHX/RANDNUM authentication)
+with-gsaapi            Enable gssapi support                       true
+                       (required for Kerberos V authentication)
+with-kerberos          Enable Kerberos V API                       true
+                       (optional for Kerberos V authentication)
+with-krb5-uam          Compile Kerberos V UAM                      true
+with-largefile         Enable largefile support                    true
+with-ldap              Enable LDAP support                         true
+with-libiconv          Enable iconv support                
+with-manual            Compile Netatalk html documentation         false
+with-overwrite         Overwrite existing configuration files      false
+with-pam               Enable PAM support                          true
+with-pgp-uam           Compile PGP UAM                             false
+with-quota             Enable Quota support                        true
+with-sendfile          Enable sendfile support                     true
+with-shadow            Enable shadow password support              true
+with-overwrite         Overwrite existing configuration files      false
+with-spotlight         Enable Spotlight support                    true
+with-tcp-wrappers      Enable TCP wrappers                         true
+with-tests             Compile tests                               false
+with-zeroconf          Enable Zeroconf (Bonjour) support           true
+```
```

Comment:
My own personal preference is to not overdocument something that is already self-documented (i.e. in meson_options.txt) – keeps maintenance overhead lower when options get changed.

I can see the argument for overdocumentation here though: To help onboard users with the new build system. Maybe we keep it like this for now, and downsize the documentation in future major releases. :)

Reply:
I agree.